### ISSUE TITLE
Some questions

### ISSUE THREAD

**Issue by boxed** [EvanKepner/mutatest#1]
(Sorry about doing this in an issue, but you didn't have an email address listed)

First of all: great to see some more interest around mutation testing! 

What is the rationale for creating a new mutation tester? There are already three mutation testers for python (I'm the author of mutmut), so if you have some use case that the existing programs doesn't cover I'd love to hear about that! I'd love to hear if you have some cool ideas or plans.

**Comment by EvanKepner** [comment]
Great to meet another tool builder! `mutatest` is mostly an excuse for me to get familiar with how mutation testing works as well as learn more about the Python AST and cache process. I was recently introduced to the mutation testing concept, and actually read [your medium article](https://hackernoon.com/mutmut-a-python-mutation-testing-system-9b9639356c78). I also took a quick glance as `cosmic-ray` as well as `mutpy` but haven't directly tested them. I will bashfully admit I haven't tried `mutmut` myself yet, but will certainly check it out. 

I am happy to keep exchanging ideas as the projects evolve. `mutatest` is in the hobby/labor-of-love camp for me but complete enough to be in the ecosystem if others are looking for ideas. The special features in architecture choices (which may or may not be unique) are:

* I wanted to work in only standard library Python, mostly as a personal challenge to myself. That said, this is hooked in to Python 3.7 features and won't work on earlier versions.
* Given the potential long running nature of trials I'm relying on random sampling of both mutation locations, as well as mutations to apply, as a trade of for the current lack of parallel trial execution.

Thanks again for reaching out. I'm looking forward to seeing how `mutmut` grows, and hopefully how others start to contribute different approaches to mutation testing to build a strong ecosystem.

**Comment by boxed** [comment]
I don't do parallel execution in mutmut, instead I've focused on making it remember what mutants it has tried even across code changes. I think it's ok-ish for most reasonable use cases now. It'd be cool to make it faster, but it's not a deal breaker. To me the deal breaker for a system like this is how hard it is to get up and running and generally to work with the tool, so that's what I've focused on with mutmut.

Do checkout mutmuts source! It's surprisingly small and readable.
